hack/local-setup-mgc.sh

```diff
@@ -0,0 +1,109 @@
+source "${LOCAL_SETUP_DIR}"/.argocdUtils
+source "${LOCAL_SETUP_DIR}"/.cleanupUtils
+source "${LOCAL_SETUP_DIR}"/.deployUtils
+
+INGRESS_NGINX_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/ingress-nginx
+METALLB_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/metallb
+CERT_MANAGER_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/cert-manager
+EXTERNAL_DNS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/external-dns
+ARGOCD_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/argocd
+ISTIO_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/istio/istio-operator.yaml
+GATEWAY_API_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/gateway-api
+REDIS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/kuadrant/redis
+THANOS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/thanos
+PROMETHEUS_FOR_FEDERATION_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/prometheus-for-federation
+GRAFANA_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/grafana
+
+set -e pipefail
+
+cleanupMGC
+
+# Deploy the submariner broker to cluster 1
+deploySubmarinerBroker ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Join cluster 1 to the submariner broker
+joinSubmarinerBroker ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Install the Gateway API CRDs in the control cluster
+installGatewayAPI ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy ingress controller
+deployIngressController ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy cert manager
+deployCertManager ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy argo cd
+deployArgoCD ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy Dashboard
+deployDashboard $KIND_CLUSTER_CONTROL_PLANE 0
+
+# Add the control plane cluster
+argocdAddCluster ${KIND_CLUSTER_CONTROL_PLANE} ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Initialize local dev setup for the controller on the control-plane cluster
+initController ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy OCM hub
+deployOCMHub ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy Redis
+deployRedis ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy MetalLb
+deployMetalLB ${KIND_CLUSTER_CONTROL_PLANE} ${metalLBSubnetStart}
+
+# Deploy Prometheus in the hub too
+deployPrometheusForFederation ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy Thanos components in the hub
+deployThanos ${KIND_CLUSTER_CONTROL_PLANE}
+
+# Deploy to workload clusters if MGC_WORKLOAD_CLUSTERS_COUNT environment variable is set
+if [[ -n "${MGC_WORKLOAD_CLUSTERS_COUNT}" ]]; then
+  for ((i = 1; i <= ${MGC_WORKLOAD_CLUSTERS_COUNT}; i++)); do
+    joinSubmarinerBroker ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployIstio ${KIND_CLUSTER_WORKLOAD}-${i}
+    installGatewayAPI ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployIngressController ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployMetalLB ${KIND_CLUSTER_WORKLOAD}-${i} $((${metalLBSubnetStart} + ${i}))
+    deployOLM ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployDashboard ${KIND_CLUSTER_WORKLOAD}-${i} ${i}
+    argocdAddCluster ${KIND_CLUSTER_CONTROL_PLANE} ${KIND_CLUSTER_WORKLOAD}-${i}
+    deployAgentSecret ${KIND_CLUSTER_WORKLOAD}-${i} "true"
+    deployAgentSecret ${KIND_CLUSTER_WORKLOAD}-${i} "false"
```

Comment:
Might be a task for after, but i don't believe these are required any longer. They were used by the old agent that no longer exists.

Reply:
Since this was existing behaviour we do not need to look into this as part of this PR, but we should make sure we have follow up issues to look into removing this.

Reply:
created #413 to revisit this part